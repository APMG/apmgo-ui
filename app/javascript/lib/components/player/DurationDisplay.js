// @flow
import React from 'react'
import './AudioPlayer.scss'

export default function (props: {duration?: number}) {
  let { duration } = props

  if (!duration || isNaN(props.duration)) {
    return (<span styleName="duration">0:00</span>)
  }

  let timeInSeconds = Math.round(duration)

  let formattedTime = ''
  let formattedMinutes = ''
  let formattedSeconds = ''
  let hours = Math.floor(timeInSeconds / 3600)
  let minutes = Math.floor((timeInSeconds / 60) - (hours * 60))
  let seconds = timeInSeconds - (minutes * 60) - (hours * 3600)

  if (hours !== 0) {
    formattedTime = hours + ':'

    if (minutes < 10) {
      formattedMinutes = '0' + minutes
    } else {
      formattedMinutes = minutes.toString()
    }
  } else {
    formattedMinutes = minutes.toString()
  }

  if (seconds < 10) {
    formattedSeconds = '0' + seconds
  } else {
    formattedSeconds = seconds.toString()
  }

  formattedTime = formattedTime + formattedMinutes + ':' + formattedSeconds

  return (<span styleName="duration">{formattedTime}</span>)
}
