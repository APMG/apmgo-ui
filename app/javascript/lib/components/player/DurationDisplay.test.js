// @flow
import React from 'react'
import { getSnapshotJson } from '../../__tests__/testHelpers'
import DurationDisplay from './DurationDisplay'

describe('Duration Time Display Component', () => {
  describe('Presenter', () => {
    it('Works Without duration Prop and Displays 0:00', () => {
      let tree = getSnapshotJson(<DurationDisplay />)
      expect(tree).toMatchSnapshot()
    })
    it('Displays duration When Prop is Passed', () => {
      let tree = getSnapshotJson(<DurationDisplay currentTime={60} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
