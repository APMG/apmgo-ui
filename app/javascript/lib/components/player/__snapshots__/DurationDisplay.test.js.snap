// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Duration Time Display Component Presenter Displays duration When Prop is Passed 1`] = `
<span
  className="AudioPlayer__duration___2Ss-f"
>
  0:00
</span>
`;

exports[`Duration Time Display Component Presenter Works Without duration Prop and Displays 0:00 1`] = `
<span
  className="AudioPlayer__duration___2Ss-f"
>
  0:00
</span>
`;
